caikit/runtime/utils/servicer_util.py

```diff
@@ -65,8 +67,19 @@ def validate_caikit_library_class_method_exists(caikit_library_class, method_nam
         raise e
 
 
-def build_proto_response(caikit_library_response):
+def build_proto_response(
+    caikit_library_response: Union[DataBase, Iterable[DataBase]]
+) -> Union[ProtoMessageType, Iterator[ProtoMessageType]]:
+    """Either serializes a data model instance into a protobuf message, or returns a lazy iterator
+    that serializes each data model from an iterable of data models"""
     try:
+        if caikit.core.isiterable(caikit_library_response):
+            # probably a streaming rpc!
```

Comment:
Can we be more certain than `probably` here? Seems like this information is known explicitly when the handler gets bound

Reply:
yeah, I'll write something better